module/zfs/dmu_recv.c

```diff
@@ -1337,8 +1342,7 @@ dmu_recv_begin(const char *tofs, const char *tosnap,
 
 	drba.drba_origin = origin;
 	drba.drba_cookie = drc;
-	drba.drba_cred = CRED();
-	drba.drba_proc = curproc;
+	drba.drba_cred = drc->drc_cred;
```

Comment:
Above and below I see plenty of error returns, that I suspect may not drop the reference.  And as I can see the caller does nothing to credentials in case of error.

Reply:
Hrm, I missed what was happening there, these are fixed up now.